src/App.tsx

```diff
@@ -14,11 +14,126 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+export enum SortBy {
+  default = '',
+  name = 'name',
+  length = 'length',
+}
+
+type State = {
+  started: boolean,
+  goods: string[],
+  sortBy: SortBy,
+  reverse: boolean,
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    started: false,
+    goods: [...goodsFromServer],
+    sortBy: SortBy.default,
+    reverse: false,
+  };
+
+  sortByName = () => {
+    this.setState({
+      sortBy: SortBy.name,
+    });
+  };
+
+  sortByLength = () => {
+    this.setState({
+      sortBy: SortBy.length,
+    });
+  };
+
+  reverse = () => {
+    this.setState(state => ({
+      reverse: !state.reverse,
+    }));
+  };
+
+  reset = () => {
+    this.setState({ goods: [...goodsFromServer], sortBy: SortBy.default });
+  };
+
+  render() {
+    const { goods, sortBy, reverse } = this.state;
+
+    const sortedGoods = [...goods].sort((g1, g2) => {
+      switch (sortBy) {
+        case 'name':
+          return g1.localeCompare(g2);
+        case 'length':
+          return g1.length - g2.length;
+        default:
+          return 0;
+      }
+    });
+
+    if (reverse) {
+      sortedGoods.reverse();
+    }
```

Comment:
вынеси это в отдельный метод
```suggestion
    const sortedGoods = this.getSortedGoods();
```

Reply:
Done!